### PR TITLE
[CodeHealth] Enable `-Wunsafe-buffer-usage` across brave

### PR DESCRIPTION
**Pull request by cdesouza-chromium**
This PR removes `brave/` from the blanket exclusion for `-Wunsafe-buffer-usage`, and adds only certain third party subpaths to it. Additionally, in order correct the ensuing warning violations, this PR adds to individual files exclusion annotations, that we can revisit gradually.

Having this warning on will prevent further violations from popping up in new places in the codebase. A significant amount of effort has already been employed in correcting some of these unsafe buffer usages.

For reference:
https://chromium.googlesource.com/chromium/src/+/refs/heads/main/docs/unsafe_buffers.md

Resolves https://github.com/brave/brave-browser/issues/41660

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: chromium_src/net/socket/socks5_client_socket.cc [hunk 1]
```diff
@@ -112,9 +113,9 @@ int SOCKS5ClientSocketAuth::Authenticate(
         DCHECK_EQ(OK, rv);
         DCHECK_LT(0u, buffer_left_);
         iobuf_ = base::MakeRefCounted<IOBufferWithSize>(buffer_left_);
-        memcpy(iobuf_->data(),
-               &buffer_.data()[buffer_.size() - buffer_left_],
-               buffer_left_);
+        UNSAFE_TODO(memcpy(iobuf_->data(),
+                           &buffer_.data()[buffer_.size() - buffer_left_],
+                           buffer_left_));
```
[the remark this answers is not in the document]
Comment (cdesouza-chromium) [reply chromium_src/net/socket/socks5_client_socket.cc:118]: The change here is mechanical to suppress a safety warning I'm enabling, which will provide some tracking for places like this, but at the moment this PR should be immaterial to the code as it is.

File: components/brave_news/browser/initialization_promise.cc [hunk 2]
```diff
@@ -94,9 +94,10 @@ void InitializationPromise::OnGotLocale(const std::string& locale) {
 
   // Determine how long we should wait based on the number of times we've
   // retried.
-  int retry_delay = no_retry_delay_for_testing_
-                        ? 0
-                        : kBackoffs[std::min(attempts_, kNumBackoffs - 1)];
+  int retry_delay =
+      no_retry_delay_for_testing_
+          ? 0
+          : UNSAFE_TODO(kBackoffs[std::min(attempts_, kNumBackoffs - 1)]);
```
[the remark this answers is not in the document]
Comment (cdesouza-chromium) [reply components/brave_news/browser/initialization_promise.cc:100]: Same

**Review by tmancey — APPROVED** [review]
LGTM

**Review by supermassive — APPROVED** [review]
wallet core lgtm

**Review by fmarier — APPROVED** [review]
Tor changes :+1:

**Review by AlexeyBarabash — APPROVED** [review]
sync ++

**Review by iefremov — APPROVED** [review]

**Review by darkdh — APPROVED** [review]

**Review by szilardszaloki — APPROVED** [review]
Playlist & Rewards LGTM.

**Review by StephenHeaps — APPROVED** [review]

**Review by kylehickinson — APPROVED** [review]

**Review by Brandon-T — APPROVED** [review]

**Pull request merged**